Added and fixed methods in Pokemon.java

gionata-bisciari opened the pull request:
**Changes made:**
- Fixed Pokemon.canEvolve():boolean, now it checks if the pokemon already reached the maximum evolution form;
- Added Pokemon.canPowerUp(considerMaxCPLimitForPlayerLevel:boolean):boolean so now passing true it checks if the pokemon is already at its current maximum cp for player level;

**Note:**
I added a new canPowerUp method (and not modified the original one) because of issue #622, it will work better after the resolution, so for now I implemented it as an "optional";

gionata-bisciari committed:
Fixed Pokemon.canEvolve();

* Fixed Pokemon.canEvolve():boolean , now checks if the pokemon already
reached the maximum evolution

gionata-bisciari committed:
Added Pokemon.canEvolve(boolean):boolean

+ Added
Pokemon.canPowerUp(considerMaxCPLimitForPlayerLevel:boolean):boolean so
now passing true it checks if the pokemon is already at its current
maximum cp for player level
* It now doesn't work perfectly because of issue #622, it will work
better after the resolution of it;

FabianTerhorst commented on .gitignore:
```diff
@@ -158,3 +158,4 @@ fabric.properties
 .LSOverride
 
 
+*.checkstyle
```
is there a reason for adding this to the gitignore?

gionata-bisciari replied:
I installed a plugin on eclipse that checks style in real time, it generated a .checkstyle file in the directory, so I added .checkstyle files to gitignore

FabianTerhorst replied:
take a look at http://stackoverflow.com/questions/1753070/git-ignore-files-only-locally

gionata-bisciari replied:
Done!

FabianTerhorst commented on library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java:
```diff
@@ -179,14 +179,35 @@ public boolean canPowerUp() throws LoginFailedException, RemoteServerException {
 	 */
 	public boolean canEvolve() throws LoginFailedException, RemoteServerException {
-		return getCandy() >= getCandiesToEvolve();
+		return ((!EvolutionInfo.isFullyEvolved(this.getPokemonId())) && (getCandy() >= getCandiesToEvolve()));
```
you can simplify the return to this: 

```
return !EvolutionInfo.isFullyEvolved(getPokemonId()) && getCandy() >= getCandiesToEvolve();
```

gionata-bisciari committed:
Removed useless brackets

Removed useless brackets

gionata-bisciari committed:
Revert "Added Pokemon.canEvolve(boolean):boolean"

This reverts commit c06efabfe8ca102e6b06cf939ef00bd1b4cf98f9.

gionata-bisciari committed:
Added Pokemon.canEvolve(boolean):boolean

+ Added
Pokemon.canPowerUp(considerMaxCPLimitForPlayerLevel:boolean):boolean so
now passing true it checks if the pokemon is already at its current
maximum cp for player level
* It now doesn't work perfectly because of issue #622, it will work
better after the resolution of it;

FabianTerhorst commented on library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java:
```diff
@@ -179,14 +179,35 @@ public boolean canPowerUp() throws LoginFailedException, RemoteServerException {
+	 * @throws RemoteServerException the remote server exception
+	 * @throws NoSuchItemException   If the PokemonId value cannot be found in the {@link PokemonMetaRegistry}.
+	 */
+	public boolean canPowerUp(boolean considerMaxCPLimitForPlayerLevel)
```
can be simplified to 

```
public boolean canPowerUp(boolean considerMaxCPLimitForPlayerLevel)
            throws LoginFailedException, RemoteServerException, NoSuchItemException {
        return considerMaxCPLimitForPlayerLevel
                ? this.canPowerUp() && (this.getCp() < this.getMaxCpForPlayer())
                : canPowerUp();
    }
```

gionata-bisciari committed:
Semplified canEvolve(boolean):boolean

*Semplified canEvolve(boolean):boolean as @FabianTerhorst suggested

gionata-bisciari committed:
Corrected indentation

FabianTerhorst commented:
still wrong indent
Its now this:

```
return considerMaxCPLimitForPlayerLevel
                ? this.canPowerUp() && (this.getCp() < this.getMaxCpForPlayer()) 
                        : canPowerUp();
```

but should be this

```
return considerMaxCPLimitForPlayerLevel
            ? this.canPowerUp() && (this.getCp() < this.getMaxCpForPlayer()) 
            : canPowerUp();
```

I can´t show you the indent with the github comments, because there are not using tabs. Intellij for example is doing it right.

gionata-bisciari commented:
public boolean canPowerUp(boolean considerMaxCPLimitForPlayerLevel)
			throws LoginFailedException, RemoteServerException, NoSuchItemException {
		return considerMaxCPLimitForPlayerLevel ? this.canPowerUp() && (this.getCp() < this.getMaxCpForPlayer())
				: canPowerUp();
	}

gionata-bisciari commented:
...
public boolean canPowerUp(boolean considerMaxCPLimitForPlayerLevel)
			throws LoginFailedException, RemoteServerException, NoSuchItemException {
		return considerMaxCPLimitForPlayerLevel ? this.canPowerUp() && (this.getCp() < this.getMaxCpForPlayer())
				: canPowerUp();
	}
...

gionata-bisciari commented:
I'm going into paranoia, have not slept in three days, I will correct later.

gionata-bisciari commented:
Last question, why all these errors pass always all tests?

FabianTerhorst commented:
because travis is ignoring checkStyle and javadoc.

gionata-bisciari commented:
but I alwais run ./gradlew checkstyleMain before commits and it pass every time

FabianTerhorst commented:
maybe checkStyle main don´t have a rule for short if.

FabianTerhorst commented:
A good ide would indent them correct.

gionata-bisciari commented:
I'm using eclipse neon, what would you recommend me?

AbandonedCart commented:
The IDE isn't really important. What matters is the formatting settings in it.
Most will support changing between 4 spaces and tab, for example.
http://stackoverflow.com/questions/1601793/how-do-i-modify-eclipse-code-formatting

FabianTerhorst commented:
@gionata-bisciari You are using cmd + shift + F to format the code?

gionata-bisciari commented:
No I'm not, it also change lines that I did not write

yutamago commented:
> I found the option under Window->Preferences->Java->Code Style->Formatter, then clicked the "Edit.." button.

From the link LoungeKatt provided.

There you can switch between **Tabs only** and **Spaces only**.
After you switched to **Tabs only** you can **ctrl + shift + f** (Source -> Format) in the file with the wrong indentation.

gionata-bisciari commented:
I always use tab only

FabianTerhorst commented:
we can also just merge this and i indent it.

gionata-bisciari commented:
I'll do it I just woke up

FabianTerhorst commented:
ok^^

gionata-bisciari commented:
Before another useless commit, is it ok?

```
public boolean canPowerUp(boolean considerMaxCPLimitForPlayerLevel)
            throws LoginFailedException, RemoteServerException, NoSuchItemException {
        return considerMaxCPLimitForPlayerLevel ? this.canPowerUp() && (this.getCp() < this.getMaxCpForPlayer())
                : canPowerUp();
    }
```

FabianTerhorst commented:
its much more readable when you indent the short if like this: 

if condition
  ? then
  : else

gionata-bisciari commented:
```
    public boolean canPowerUp(boolean considerMaxCPLimitForPlayerLevel)
            throws LoginFailedException, RemoteServerException, NoSuchItemException {
        return considerMaxCPLimitForPlayerLevel 
                ? this.canPowerUp() && (this.getCp() < this.getMaxCpForPlayer())
                : canPowerUp();
    }
```

now?

FabianTerhorst commented:
yes

gionata-bisciari committed:
Indentation fix
